Engine/Platform/System/Posix/CYBPosixFile.cpp

```diff
@@ -0,0 +1,160 @@
+	}
+
+	if (AMethod == API::File::Method::ANY)
+		AMethod = API::File::Method::CREATE;
+}
+ 
+CYB::Platform::System::File::File(Path&& APath, const Mode AMode, const Method AMethod) :
+	FPath(std::move(APath)),
+	FOpenMode(AMode),
+	FOpenMethod(AMethod)
+{
+	Init(FPath, AMode, FOpenMethod);
+}
+
+CYB::Platform::System::Implementation::File::File(File&& AMove) noexcept :
+	FDescriptor(AMove.FDescriptor) {
+	AMove.FDescriptor = -1;
+}
+
+CYB::Platform::System::File& CYB::Platform::System::File::operator=(File&& AMove) noexcept {
+	Close();
+	FDescriptor = AMove.FDescriptor;
+	AMove.FDescriptor = -1;
+	FOpenMode = AMove.FOpenMode;
+	FOpenMethod = AMove.FOpenMethod;
+	return *this;
+}
+
+void CYB::Platform::System::Implementation::File::Close(void) const noexcept {
+	if (FDescriptor != -1)
+		Core().FModuleManager.FC.Call<Modules::LibC::close>(FDescriptor);
+}
+
+CYB::Platform::System::File::~File() {
+	Close();
+}
+
+unsigned long long CYB::Platform::System::File::Size(const System::Path& APath) {
+	StatStruct Stat;
+	const auto Result(Sys::Call(Sys::LSTAT, const_cast<char*>(APath().CString()), &Stat));
+	if(Result == 0)
+		return static_cast<unsigned long long>(Stat.st_size);
+	else if(Result == EACCES)
+		throw Exception::SystemData(Exception::SystemData::FILE_NOT_READABLE);
+	throw Exception::SystemData(Exception::SystemData::FILE_NOT_FOUND);
+}
+
+
+StatStruct CYB::Platform::System::Implementation::File::StatFD(void) const noexcept {
+	StatStruct Stat;
+	const auto Result(static_cast<int>(System::Sys::Call(Sys::FSTAT, FDescriptor, &Stat)));
+	API::Assert::Equal(Result, static_cast<decltype(Result)>(0));
+	return Stat;
+}
+
+unsigned long long CYB::Platform::System::File::Size(void) const noexcept {
+	return static_cast<unsigned long long>(StatFD().st_size);
+}
+
+unsigned long long CYB::Platform::System::File::Seek(const long long AOffset, const SeekLocation ALocation) const {
+	const auto PosixLocation([&]() {
+		switch (ALocation) {
+		case SeekLocation::BEGIN:
+			return SEEK_SET;
+		case SeekLocation::CURSOR:
+			return SEEK_CUR;
+		case SeekLocation::END:
+			return SEEK_END;
+		default:
+			throw Exception::Violation(Exception::Violation::INVALID_ENUM);
+		}
+	}());
+
+	const auto Result(Core().FModuleManager.FC.Call<Modules::LibC::lseek>(FDescriptor, static_cast<off_t>(AOffset), PosixLocation));
+	API::Assert::NotEqual(Result, static_cast<decltype(Result)>(-1));
+
+	return static_cast<unsigned long long>(Result);
+}
+
+unsigned long long CYB::Platform::System::File::Read(void* const ABuffer, const unsigned long long AMaxAmount) const {
+	if (FOpenMode == Mode::WRITE)
+		throw Exception::Violation(Exception::Violation::INVALID_OPERATION);
+	const auto Result(Core().FModuleManager.FC.Call<Modules::LibC::read>(FDescriptor, ABuffer, static_cast<size_t>(AMaxAmount)));
+	API::Assert::NotEqual(Result, static_cast<decltype(Result)>(-1));
+	return static_cast<unsigned long long>(Result);
+}
```

Comment:
These will need to be refactored into not being declared in API::File, but rather API::(Intefaces::?)::Stream or something alone those lines